index.js

```diff
@@ -345,24 +347,41 @@ Client.prototype._encode = function (obj, enc) {
   return ndjson.serialize(out)
 }
 
+Client.prototype._flushCallsWaitingMetadata = function () {
+  for (const args of this._callsWaitingMetadata) {
+    this.write(...args)
+  }
+
+  this._callsWaitingMetadata = []
+}
+
+Client.prototype._sendOrAddToWaitingMetadata = function(...args) {
+  if (this._hasSentMetadata) {
+    this._maybeUncork()
+    return this.write(...args)
+  } else {
+    if (this._callsWaitingMetadata.length === 0) {
+      this._chopper.write('')
+    }
+    this._callsWaitingMetadata.push(args)
+    return true
```

Comment:
This would make the return value for `sendSpan`, `sendTransaction`, etc to always be true when waiting for the metadata to be written, this has no implications on the current code base as those return values are ignored.